src/lib.rs

```diff
@@ -6,6 +6,8 @@ use core::{
 use hex::FromHex;
 use http::uri::{Authority, InvalidUri};
 use iri_string::types::UriString;
+use serde::de::Visitor;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
```

Comment:
I personally prefer when it's tidy like this 🙈 

```suggestion
use serde::{de::{self, Visitor}, Deserialize, Deserializer, Serialize, Serializer};
```